mongodump/mongodump_test.go

```diff
@@ -2091,3 +2104,86 @@ func TestMongoDumpColumnstoreIndexes(t *testing.T) {
 		require.Equal(t, count, 1)
 	}
 }
+
+func TestOptionsOrderIsPreserved(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	log.SetWriter(io.Discard)
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	require.NoError(t, err)
+
+	collName := "students"
+	viewName := "studentsView"
+
+	pipeline := bson.A{
+		bson.D{{Key: "$group", Value: bson.D{
+			{Key: "_id", Value: bson.D{
+				{Key: "year", Value: "$year"},
+				{Key: "name", Value: "$name"},
+			}},
+			{Key: "highest", Value: bson.D{
+				{Key: "$max", Value: "$score"},
+			}},
+		}}},
+		bson.D{{Key: "$project", Value: bson.D{
+			{Key: "_id", Value: 0},
+		}}},
+		bson.D{{Key: "$sort", Value: bson.D{
+			{Key: "year", Value: 1},
+			{Key: "sID", Value: -1},
+			{Key: "name", Value: 1},
+			{Key: "score", Value: 1},
+		}}},
+	}
+
```

Comment:
Given that the original bug of reordering is probabilistic, can we shuffle the pipeline in this test to test different orderings?